packages/cli/src/commands/pg/wait.ts

```diff
@@ -0,0 +1,93 @@
+import color from '@heroku-cli/color'
+import {Command, flags} from '@heroku-cli/command'
+import {Args, ux} from '@oclif/core'
+import debug from 'debug'
+import {all, getAddon} from '../../lib/pg/fetcher'
+import pgHost from '../../lib/pg/host'
+import notify from '../../lib/notify'
+import {AddOnAttachmentWithConfigVarsAndPlan, AddOnWithRelatedData, PgStatus} from '../../lib/pg/types'
+import {HTTPError} from 'http-call'
+
+const wait = (ms: number) => new Promise(resolve => {
+  setTimeout(resolve, ms)
+})
+
+export default class Wait extends Command {
+  static topic = 'pg';
+  static description = 'blocks until database is available';
+  static flags = {
+    'wait-interval': flags.string({description: 'how frequently to poll in seconds (to avoid rate limiting)'}),
+    'no-notify': flags.boolean({description: 'do not show OS notification'}),
+    app: flags.app({required: true}),
+    remote: flags.remote(),
+  };
+
+  static args = {
+    database: Args.string(),
+  };
+
+  public async run(): Promise<void> {
+    const {flags, args} = await this.parse(Wait)
+    const {app, 'wait-interval': waitInterval} = flags
+    const dbName = args.database
+    const pgDebug = debug('pg')
+
+    const waitFor = async (db: AddOnAttachmentWithConfigVarsAndPlan | AddOnWithRelatedData) => {
+      let interval = waitInterval && Number.parseInt(waitInterval, 10)
+      if (!interval || interval < 0) interval = 5
+      let status
+      let waiting = false
+      let retries = 20
+
+      while (true) {
+        try {
+          status = await this.heroku.get<PgStatus>(
+            `/client/v11/databases/${db.id}/wait_status`,
+            {
+              hostname: pgHost(),
+            })
+            .then(response => response.body)
```

Comment:
This I learnt from Justin, it's a good trick and avoids the _thenable_ function usage:

```suggestion
          ({body: status} = await this.heroku.get<PgStatus>(
            `/client/v11/databases/${db.id}/wait_status`,
            {hostname: pgHost()},
          ))
```

Reply:
Oh nice! Thanks! I made the update.